### PR TITLE
 Call scroll to when scroll view offset changes

### PR DESCRIPTION
**Pull request by mstach60161**
## Summary

This PR adds additional functionality to `scrollViewOffset` prop - #4414.
When we modify value of `scrollViewOffset`, the `scrollView` component moves to given position.

It's done by adding listener to `scrollViewOffset` and whenever the value is changed, listener calls `scrollTo` method.

To prevent cycle, I added an additional property to the object - `triggerScrollListener`. When the value is changed inside the event, it indicates that scrolling has already occurred, or the user has manually dragged the ScrollView. In this case, we want to trigger mappers, but we don't want to run the listener responsible for scrolling.

Additionally, to avoid triggering mappers twice, I added a field - `triggerOffsetEvent`. When the event is initiated by the `scrollTo` inside listener, it means that the value of scrollViewOffset has already been updated, and we don't want to trigger mappers again, by changing value inside the event.

## Test plan

- Example app

### CODE REVIEW

File: app/src/examples/ScrollViewOffsetExample.tsx [hunk 1]
```diff
@@ -2,45 +2,77 @@ import React from 'react';
...
+  withTiming,
+  useAnimatedReaction,
+  scrollTo,
   useAnimatedStyle,
   useSharedValue,
 } from 'react-native-reanimated';
 import { Button, StyleSheet, Text, View } from 'react-native';
 
-export default function ScrollViewOffsetExample() {
+export default function ScrollViewOffsetCallScrollToExample() {
```
[the remark this answers is not in the document]
Comment (mstach60161) [reply app/src/examples/ScrollViewOffsetExample.tsx:13]: Yes, that was the idea, but I accidentally got rid of the previous one. 
Thanks!

**Comment by henrymoulton** [comment]
Hey @mstach60161 are you still working on this PR? I'm keen to take a look or help based off discussion here: also ripped out flipper as android prod builds fail as well

**Comment by henrymoulton** [comment]
Hey @mstach60161 are you still working on this PR? I'm keen to take a look or help based off discussion here: https://twitter.com/HenryMoulton/status/1729470321290342511

**Comment by tomekzaw** [comment]
@henrymoulton Michał is no longer working on it, you can go ahead and take a look if you want

**Comment by piaskowyk** [comment]
At the current time, we no longer intend to expand the public API in this field. However, we will retain these changes for the future. 🫡

**Pull request closed without merging**